utils/splunk/splunk_telemetry_base.py

```diff
@@ -38,7 +66,24 @@ def check(self, instance):
         instance.update_status(current_time, self.status)
 
         try:
-            self._auth_session(instance)
+            if authentication and 'token_auth' in authentication:
```

Comment:
This is duplicate code; here and in splunk_topology/check.py.

Can we create a similar setup for token auth as we did for basic auth, in the helper class, `self._auth_session(instance)`